org.eclipse.lsp4e.jdt/.settings/org.eclipse.jdt.core.prefs

```diff
@@ -9,7 +9,7 @@ org.eclipse.jdt.core.compiler.annotation.nonnullbydefault.secondary=
 org.eclipse.jdt.core.compiler.annotation.notowning=org.eclipse.jdt.annotation.NotOwning
 org.eclipse.jdt.core.compiler.annotation.nullable=org.eclipse.jdt.annotation.Nullable
 org.eclipse.jdt.core.compiler.annotation.nullable.secondary=
-org.eclipse.jdt.core.compiler.annotation.nullanalysis=enabled
+org.eclipse.jdt.core.compiler.annotation.nullanalysis=disabled
```

Comment:
would you not prefer to follow the other projects in this settings? What is the reason for disabling it?

Reply:
I would... but I'm lost with JDT Null check annotations... I see that LSP4E plugin project has lots of errors coming from JDT related to Null checks but the maven build goes through. The `lsp4e.jdt` project also has a bunch of same origin errors however maven build fails with compilation errors for the JDT null check issues... I wasn't able to figure out where the difference between `lsp4e` and `lsp4e.jdt` projects comes from... Disabling the null checking in the preference solved it for the `lsp4e.jdt` for me...

Reply:
It seems that I need to add `@NonNull` annotations in JDT UI in `ISemanticTokensProvider` and the plugin there depend on `org.eclipse.jdt.annotation` plugin... Otherwise without ignoring null violation analysis i can't get it to compile :-\

Reply:
Sorry for the long delay.

You can annotate libraries externally with annotations, in LSP4E we use the library https://github.com/vegardit/no-npe instead of doing it directly in our repo. You should be able to fix it by adding annotations in https://github.com/vegardit/no-npe/tree/main/libs/eea-eclipse-platform/src/main/resources/org/eclipse

Reply:
Hmm... I didn't see `eaa-lsp4e` in there... Or did you mean I should add `lsp4e` in there?